test/e2e/functional/tests/e2e.go

```diff
@@ -73,47 +74,64 @@ var _ = Describe("metallb", func() {
 
 		It("should have MetalLB pods in running state", func() {
 			By("checking MetalLB controller deployment is in running state", func() {
-				Eventually(func() bool {
+				Eventually(func() error {
 					deploy, err := testclient.Client.Deployments(metallb.Namespace).Get(context.Background(), consts.MetalLBDeploymentName, metav1.GetOptions{})
 					if err != nil {
-						return false
+						return err
 					}
-					return deploy.Status.ReadyReplicas > 0 && deploy.Status.ReadyReplicas == deploy.Status.Replicas
-				}, metallbutils.DeployTimeout, metallbutils.Interval).Should(BeTrue())
 
-				pods, err := testclient.Client.Pods(OperatorNameSpace).List(context.Background(), metav1.ListOptions{
-					LabelSelector: "component=controller"})
-				Expect(err).ToNot(HaveOccurred())
+					if deploy.Status.ReadyReplicas <= 0 || deploy.Status.ReadyReplicas != deploy.Status.Replicas {
+						return pkgerrors.New("controller deployment is not ready")
```

Comment:
can this be more informative?
something like:
`fmt.Errorf("deployment %s is not ready, expected %d ready replicas got %d",...)`